sycl/plugins/level_zero/pi_level_zero.cpp

```diff
@@ -3066,12 +3066,19 @@ pi_result piKernelGetInfo(pi_kernel Kernel, pi_kernel_info ParamName,
   case PI_KERNEL_INFO_PROGRAM:
     return ReturnValue(pi_program{Kernel->Program});
   case PI_KERNEL_INFO_FUNCTION_NAME:
-    // TODO: Replace with the line in the comment once bug in the Level Zero
-    // driver will be fixed. Problem is that currently Level Zero driver
-    // truncates name of the returned kernel if it is longer than 256 symbols.
-    //
-    // return ReturnValue(ZeKernelProperties.name);
-    return ReturnValue(Kernel->KernelName.c_str());
+    try {
+      size_t Size = 0;
+      ZE_CALL(zeKernelGetName(Kernel->ZeKernel, &Size, nullptr));
+      char *KernelName = new char[Size];
+      ZE_CALL(zeKernelGetName(Kernel->ZeKernel, &Size, KernelName));
```

Comment:
Can we skip this if we are only going to return the size, i.e. `ParamValue` is null?
I see that we currently are doing strlen of the string to return to re-calculate the size though, which would need to change then.

Thoughts?

Anyways, this must not be a part of this PR.